src/math2d/regioned_angle.rs

```diff
@@ -0,0 +1,111 @@
+use std::f32::consts::PI;
+
+use super::{line::Line, point::Point, vector::Vector};
+
+pub trait RegionedAngleTrait {
+    fn get_radians(&self) -> f32;
+    fn degrees(&self) -> f32;
+    fn min_degrees(&self) -> i32;
+    fn max_degrees(&self) -> i32;
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, PartialOrd)]
+pub struct RegionedAngle<const MIN_DEGREES: i32, const MAX_DEGREES: i32> {
+    radians: f32,
+}
+
+impl<const MIN_DEGREES: i32, const MAX_DEGREES: i32> RegionedAngle<MIN_DEGREES, MAX_DEGREES> {
+    pub fn new() -> Self {
+        Self { radians: 0.0 }
+    }
+
+    pub fn from_angle<A>(other: &A) -> Self
+    where
+        A: RegionedAngleTrait,
+    {
+        let mut regioned_angle: RegionedAngle<MIN_DEGREES, MAX_DEGREES> = RegionedAngle {
+            radians: other.get_radians(),
+        };
+        regioned_angle.move_to_range();
+        regioned_angle
+    }
+
+    fn move_to_range(&mut self) -> Self {
+        while self.radians < MIN_DEGREES as f32 / 180.0 * PI {
+            self.radians += 2.0 * PI;
+        }
+        while self.radians > MAX_DEGREES as f32 / 180.0 * PI {
+            self.radians -= 2.0 * PI;
+        }
+        *self
+    }
+
+    fn radians_from_vectors(v1: &Vector, v2: &Vector) -> f32 {
+        let (xx1, yy1) = v1.get_coordinates();
+        let (xx2, yy2) = v2.get_coordinates();
+        let dot_product = xx1 * xx2 + yy1 * yy2;
```

Comment:
I know I didn't do it, but it would be cool to say "let dot_product = v1.dot(v2)" here